src/common/world.cpp

```diff
@@ -1,34 +1,34 @@
 #include <voxen/common/world.hpp>
 #include <voxen/util/log.hpp>
 
+#include <glm/ext/quaternion_exponential.hpp>
+
 namespace voxen
 {
 
-World::World() : m_terrain(/*1<<21*/16, /*1 << 12*/4) {
+World::World() : m_terrain(/*1<<21*/16, /*1 << 12*/4), m_tick_id(0) {
 }
 
-World::World(const World &other) : m_player(other.m_player), m_terrain(other.m_terrain) {
+World::World(const World &other) : m_player(other.m_player), m_terrain(other.m_terrain), m_tick_id(other.m_tick_id) {
 }
 
 World::~World() {}
 
-void World::update() {
-	auto pos = m_player.position();
+void World::update(DebugQueueRtW& queue, std::chrono::duration<int64_t, std::nano> tick_inverval) {
+	// Update user position
+	double dt = std::chrono::duration_cast<std::chrono::duration<double>>(tick_inverval).count();
+	glm::dvec3 pos = m_player.position();
+	pos += queue.player_forward_movement_direction * (queue.forward_speed * dt);
+	pos += queue.player_strafe_movement_direction * (queue.strafe_speed * dt);
+	m_player.updateState(pos, queue.player_orientation);
+
 	m_terrain.updateChunks(pos.x, pos.y, pos.z);
+
+	m_tick_id++;
```

Comment:
There is a nuisance with `m_tick_id` logic - its value is `N` during `update()` execution, but then the render gets the same state with ID `N+1` because we've incremented it. This behaviour may be non-intuitive during debugging. I'd suggest to move `m_tick_id++` to the beginning of `update()` body so tick ID will be equal for both computing and rendering.